api/src/main/java/org/openmrs/module/mirebalais/setup/HtmlFormSetup.java

```diff
@@ -1,114 +1,43 @@
-                        "pihcore:htmlforms/haiti/liveCheckin.xml",
-                        "pihcore:htmlforms/haiti/hiv/zl/vct.xml",
-                        "pihcore:htmlforms/haiti/hiv/zl/section-hiv-history.xml",
-                        "pihcore:htmlforms/haiti/hiv/zl/section-symptoms.xml",
-                        "pihcore:htmlforms/haiti/hiv/zl/section-hiv-assessment.xml",
-                        "pihcore:htmlforms/haiti/hiv/zl/section-hiv-state.xml",
-                        "pihcore:htmlforms/haiti/hiv/iSantePlus/Adherence.xml",
-                        "pihcore:htmlforms/haiti/hiv/iSantePlus/SaisiePremiereVisiteAdult.xml",
-                        "pihcore:htmlforms/haiti/hiv/iSantePlus/SaisiePremiereVisitePediatrique.xml",
-                        "pihcore:htmlforms/haiti/hiv/iSantePlus/VisiteDeSuivi.xml",
-                        "pihcore:htmlforms/haiti/hiv/iSantePlus/VisiteDeSuiviPediatrique.xml"
-                ));
-            }
-            else if (config.getCountry().equals(ConfigDescriptor.Country.LIBERIA)) {
-                htmlforms.addAll(Arrays.asList(
-                        "pihcore:htmlforms/liberia/checkin.xml",
-                        "pihcore:htmlforms/liberia/liveCheckin.xml",
-                        "pihcore:htmlforms/liberia/patientRegistration-social.xml"
-                ));
-            }
+        ResourceFactory resourceFactory = ResourceFactory.getInstance();
+        FormService formService = Context.getFormService();
+        HtmlFormEntryService htmlFormEntryService = Context.getService(HtmlFormEntryService.class);
 
-            else if (config.getCountry().equals(ConfigDescriptor.Country.SIERRA_LEONE)) {
-                htmlforms.addAll(Arrays.asList(
-                        "pihcore:htmlforms/patientRegistration.xml",
-                        "pihcore:htmlforms/patientRegistration-rs.xml",
-                        "pihcore:htmlforms/sierra_leone/checkin.xml",
-                        "pihcore:htmlforms/sierra_leone/liveCheckin.xml",
-                        "pihcore:htmlforms/sierra_leone/patientRegistration-social.xml",
-                        "pihcore:htmlforms/sierra_leone/patientRegistration-ebolaScreening.xml"
-                ));
-            }
+        File htmlformDir = new File(OpenmrsUtil.getApplicationDataDirectory() + PihCoreUtil.getFormDirectory());
+        Collection<File> files = FileUtils.listFiles(htmlformDir, null, true);
 
-            if (htmlforms != null) {
-                for (String htmlform : htmlforms) {
-                    HtmlFormUtil.getHtmlFormFromUiResource(resourceFactory, formService, htmlFormEntryService, htmlform);
+        try {
+            for (File file : files) {
+                try {
+                    HtmlFormUtil.getHtmlFormFromUiResource(resourceFactory, formService, htmlFormEntryService, "file:" + file.getAbsolutePath());
+                } catch (IOException e) {
+                    log.error("Unable to load HTML Form at path: " + file, e);
```

Comment:
The decision to survive this error is something we should continue to evaluate. For now I support it. In the future it might be cool to have it create a stub form, so that an HFE XML that isn't initially valid can be iterated on using hot reloading.

Reply:
I haven't really evaluated yet the situations in which this exception might be hit, but I wanted to prevent against a stray form error completely preventing an OpenMRS startup.